### ISSUE TITLE
SVT Play cant start some programs

### ISSUE THREAD

**Issue by ToSeX** [kodi-svtplay/xbmc-svtplay#148]
For example http://www.svtplay.se/liv-och-horace-i-europa
Tested i Openelec and KODI 
Also tested to send from Air play clieint

But program like http://www.svtplay.se/video/6990301/aktuellt/aktuellt-11-mar-21-00?start=auto
works fine...

SVT Version:    4.0.21
KODI 16.0 GIT:0160220-a5f3a99
[kodi_log.txt](https://github.com/nilzen/xbmc-svtplay/files/170416/kodi_log.txt)

**Comment by tkrill** [comment]
Seems like some streams no longer play while others still seems to work. Could it be some new/unsupported stream format  used by SVT?

I get this when a stream fail to play:

13:09:28 T:139702996821760 WARNING: CreateLoader - unsupported protocol(plugin) in plugin://plugin.video.svtplay/?url=......

**Comment by ToSeX** [comment]
The spooky thing is that the streams worked fine in for example OpenELEC-Nvidia_Legacy.x86_64-6.0.3...

Br
Tobbe S

**Comment by linqcan** [comment]
Sounds platform/kodi build dependent.

Please specify steps to reproduce, including shows/streams that do not work
for you and I will have a look.

**Comment by ToSeX** [comment]
OK here we go...

[Namnlös 1.zip](https://github.com/nilzen/xbmc-svtplay/files/170906/Namnlos.1.zip)

Br
Tobbe S

**Comment by pacmyc** [comment]
Raspberry Pi 2 running Kodi.
Getting the same error here on some programs, for example "vem vet mest";

"18:02:52 1875.828979 T:1957814832  NOTICE: DVDPlayer: Opening: index_6_av.m3u8?e=82d461566017da20&alt=http%3a%2f%2fswitcher.cdn.svt.se%2f1369718-043a.m3u8&cc1=name=svenska~default=yes~forced=no~uri=http://media.svt.se/download/mcc/test/core-prd/SUB-1369718-043A-VEMVETMEST/index.m3u8~lang=sv
18:02:52 1875.829590 T:1957814832 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
18:02:52 1875.830200 T:1957814832   DEBUG: CRenderManager::UpdateDisplayLatency - Latency set to 0 msec
18:02:52 1875.830322 T:1957814832   DEBUG: CMMALRenderer::UnInitMMAL pool((nil))
18:02:52 1875.830688 T:1957814832   DEBUG: CMMALRenderer::PreInit
18:02:52 1875.831177 T:1648358384   DEBUG: Thread DVDPlayer start, auto delete: false
18:02:52 1875.832275 T:1648358384  NOTICE: Creating InputStream
18:02:52 1875.833130 T:1648358384 WARNING: CreateLoader - unsupported protocol(index_6_av.m3u8?e=82d461566017da20&alt=http%3a%2f%2fswitcher.cdn.svt.se%2f1369718-043a.m3u8&cc1=name=svenska~default=yes~forced=no~uri=http) in index_6_av.m3u8?e=82d461566017da20&alt=http%3a%2f%2fswitcher.cdn.svt.se%2f1369718-043a.m3u8&cc1=name=svenska~default=yes~forced=no~uri=http://media.svt.se/download/mcc/test/core-prd/SUB-1369718-043A-VEMVETMEST/index.m3u8~lang=sv
18:02:52 1875.833374 T:1648358384   ERROR: CDVDPlayer::OpenInputStream - error opening [index_6_av.m3u8?e=82d461566017da20&alt=http%3A%2F%2Fswitcher.cdn.svt.se%2F1369718-043A.m3u8&cc1=name=Svenska~default=yes~forced=no~uri=http://media.svt.se/download/mcc/test/core-prd/SUB-1369718-043A-VEMVETMEST/index.m3u8~lang=sv]
18:02:52 1875.833496 T:1648358384  NOTICE: CDVDPlayer::OnExit()
18:02:52 1875.833618 T:1648358384   DEBUG: OnPlayBackStopped: play state was 1, starting 1
18:02:52 1875.833740 T:1648358384   DEBUG: Thread DVDPlayer 1648358384 terminating
18:02:52 1875.833862 T:1957814832   DEBUG: OnPlayBackStopped: play state was 3, starting 0
18:02:52 1875.834839 T:1957814832   DEBUG: CAnnouncementManager - Announcement: OnStop from xbmc
18:02:52 1875.834961 T:1957814832   DEBUG: GOT ANNOUNCEMENT, type: 1, from xbmc, message OnStop
18:02:52 1875.835571 T:1957814832   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.svtplay/?url=%2Fvideo%2F6968125%2Fvem-vet-mest%2Fvem-vet-mest-sasong-16-avsnitt-43&mode=video]"

**Comment by linqcan** [comment]
Yep, spotted another issue too. Working on this now.

**Comment by linqcan** [comment]
Kodi pull requested for version 4.0.22.

**Comment by pacmyc** [comment]
I still have the same problem in v 4.0.22 on a Rasperry Pi 2 running Kodi.

**Comment by linqcan** [comment]
Are you really running v4.0.22? It hasnt gone live yet...

If so, please describe the steps needed to reproduce the issue. That is; where you are clicking in the menus and finally what program you are trying to watch.

**Comment by pacmyc** [comment]
Yes, I downloaded the ZIP-file directly to the Pi with wget https://github.com/nilzen/xbmc-svtplay/archive/gotham.zip
and then manually installed it. I also confirmed that the new version number really was installed.

**Comment by linqcan** [comment]
OK. A "works here" syndrome. I need to know the steps needed to reproduce
the issue. Also, please include a log from your system. Thanks!

**Comment by ToSeX** [comment]
I did the same and was the same symptoms as in the "root" case.

**Comment by pacmyc** [comment]
Hi.
OK, sorry for delay, I was at work.
So these are the steps;
1. Start SVT play
2. Go to Programs A-Ö
3. Go down to "Vem vet mest?"
4. Accidently hit "..."
5. Go into "Vem vet mest" again
6. Start "Säsong 16 - Avsnitt 48"
7. Nothing happens.

`Script started on Wed Mar 16 19:41:04 2016
]0;osmc@osmc: ~osmc@osmc:~$ tail -f .kodi/[K[K[K[K[K[K/.ko[K[K[K[K.kodi/
addons/   media/    sounds/   system/   temp/     userdata/ 
]0;osmc@osmc: ~osmc@osmc:~$ tail -f .kodi/
addons/   media/    sounds/   system/   temp/     userdata/ 
]0;osmc@osmc: ~osmc@osmc:~$ tail -f .kodi/temp/kodi.log
19:40:58 102.468880 T:1777345520   DEBUG: DoWork - took 144 ms to load special://masterprofile/Thumbnails/5/53dd3c95.jpg
19:40:58 102.946175 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:40:58 102.946426 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:40:59 103.234032 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:40:59 103.234344 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:29 133.109146 T:1785734128   DEBUG: script.module.osmcsetting.updates :  - blurp 182 - Home.xml
19:41:31 135.285187 T:1731195888   DEBUG: Thread JobWorker 1731195888 terminating (autodelete)
19:41:31 135.285446 T:1749545968   DEBUG: Thread JobWorker 1749545968 terminating (autodelete)
19:41:31 135.285812 T:1777345520   DEBUG: Thread JobWorker 1777345520 terminating (autodelete)
19:41:31 135.366302 T:1655698416   DEBUG: Thread JobWorker 1655698416 terminating (autodelete)
19:41:43 147.980392 T:1957675568   DEBUG: Keyboard: scancode: 0x1c, sym: 0x000d, unicode: 0x0000, modifier: 0x0
19:41:44 148.100601 T:1957675568   DEBUG: OnKey: return (0xf00d) pressed, action is Select
19:41:44 148.102463 T:1957675568   DEBUG: Activating window ID: 10025
19:41:44 148.347321 T:1957675568   DEBUG: ------ Window Deinit (Home.xml) ------
19:41:44 148.958923 T:1957675568   DEBUG: ------ Window Init (MyVideoNav.xml) ------
19:41:44 148.959122 T:1957675568    INFO: Loading skin file: MyVideoNav.xml, load type: KEEP_IN_MEMORY
19:41:45 149.279449 T:1957675568   DEBUG: CGUIMediaWindow::GetDirectory (plugin://plugin.video.svtplay/)
19:41:45 149.279724 T:1957675568   DEBUG:   ParentPath = [plugin://plugin.video.svtplay/]
19:41:45 149.283112 T:1957675568   DEBUG: Loading items: 12, directory: plugin://plugin.video.svtplay/ sort method: 0, ascending: false
19:41:45 149.296021 T:1957675568   DEBUG: RunQuery took 2 ms for 0 items query: SELECT \* FROM tvshow_view  WHERE idParentPath=7
19:41:45 149.317963 T:1957675568   DEBUG: RunQuery took 19 ms for 138 items query: SELECT  files.strFilename, files.playCount,  bookmark.timeInSeconds, bookmark.totalTimeInSeconds FROM files  LEFT JOIN bookmark ON    files.idFile = bookmark.idFile AND bookmark.type = 1  WHERE files.idPath=7
19:41:45 149.350906 T:1655698416   DEBUG: Thread BackgroundLoader start, auto delete: false
19:41:45 149.403015 T:1655698416   DEBUG: Thread BackgroundLoader 1655698416 terminating
19:41:51 155.775009 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:51 155.775330 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:52 156.210632 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:52 156.210907 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:52 156.543472 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:52 156.543747 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:52 156.918808 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:52 156.919388 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:53 157.474014 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:53 157.474564 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:53 157.849396 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:53 157.850067 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:54 158.469666 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:41:54 158.470490 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:41:56 160.060257 T:1957675568   DEBUG: Keyboard: scancode: 0x1c, sym: 0x000d, unicode: 0x0000, modifier: 0x0
19:41:56 160.180923 T:1957675568   DEBUG: OnKey: return (0xf00d) pressed, action is Select
19:41:56 160.184723 T:1957675568   DEBUG: Clearing cached fileitems [plugin://plugin.video.svtplay/?mode=a-o]
19:41:56 160.186432 T:1957675568   DEBUG: CGUIMediaWindow::GetDirectory (plugin://plugin.video.svtplay/?mode=a-o)
19:41:56 160.186859 T:1957675568   DEBUG:   ParentPath = [plugin://plugin.video.svtplay/]
19:41:56 160.188812 T:1655698416   DEBUG: Thread JobWorker start, auto delete: true
19:41:56 160.189651 T:1655698416   DEBUG: StartScript - calling plugin SVT Play('plugin://plugin.video.svtplay/','1','?mode=a-o')
19:41:56 160.190475 T:1777345520   DEBUG: Thread LanguageInvoker start, auto delete: false
19:41:56 160.190720 T:1655698416   DEBUG: WaitOnScriptResult - waiting on the SVT Play (id=4) plugin...
19:41:56 160.190857 T:1777345520    INFO: initializing python engine.
19:41:56 160.190964 T:1777345520   DEBUG: CPythonInvoker(4, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): start processing
19:41:56 160.570557 T:1777345520   DEBUG: -->Python Interpreter Initialized<--
19:41:56 160.570892 T:1777345520   DEBUG: CPythonInvoker(4, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): the source file to load is "/home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py"
19:41:56 160.571579 T:1777345520   DEBUG: CPythonInvoker(4, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): setting the Python path to /home/osmc/.kodi/addons/xbmc-svtplay-gotham:/home/osmc/.kodi/addons/script.module.parsedom/lib:/usr/lib/python2.7:/usr/lib/python2.7/plat-arm-linux-gnueabihf:/usr/lib/python2.7/lib-tk:/usr/lib/python2.7/lib-old:/usr/lib/python2.7/lib-dynload:/usr/local/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages/PILcompat:/usr/lib/python2.7/dist-packages/gtk-2.0
19:41:56 160.571747 T:1777345520   DEBUG: CPythonInvoker(4, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): entering source directory /home/osmc/.kodi/addons/xbmc-svtplay-gotham
19:41:56 160.571945 T:1777345520   DEBUG: CPythonInvoker(4, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): instantiating addon using automatically obtained id of "plugin.video.svtplay" dependent on version 2.14.0 of the xbmc.python api
19:41:56 160.688934 T:1957675568   DEBUG: ------ Window Init (DialogBusy.xml) ------
19:41:57 161.723846 T:1777345520   DEBUG: CommonFunctions-2.5.1
19:42:00 164.871933 T:1777345520   DEBUG: LocalizeStrings: no strings.po file exist at /home/osmc/.kodi/addons/xbmc-svtplay-gotham/resources/language/English, fallback to strings.xml
19:42:01 165.042496 T:1655698416   DEBUG: WaitOnScriptResult- plugin returned successfully
19:42:01 165.044296 T:1777345520    INFO: CPythonInvoker(4, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): script successfully run
19:42:01 165.066864 T:1957675568   DEBUG: Saving fileitems [plugin://plugin.video.svtplay/?mode=a-o]
19:42:01 165.075134 T:1957675568   DEBUG:   -- items: 445, sort method: 0, ascending: false
19:42:01 165.143936 T:1777345520    INFO: Python script stopped
19:42:01 165.144272 T:1777345520   DEBUG: Thread LanguageInvoker 1777345520 terminating
19:42:01 165.263840 T:1777345520   DEBUG: Thread BackgroundLoader start, auto delete: false
19:42:01 165.414963 T:1957675568   DEBUG: ------ Window Deinit (DialogBusy.xml) ------
19:42:02 166.098602 T:1777345520   DEBUG: Thread BackgroundLoader 1777345520 terminating
        <<< deleted some keypressing logs  >>>
19:42:31 195.044205 T:1655698416   DEBUG: Thread JobWorker 1655698416 terminating (autodelete)
        <<< deleted some keypressing logs  >>>
19:42:37 201.159653 T:1957675568   DEBUG: Keyboard: scancode: 0x1c, sym: 0x000d, unicode: 0x0000, modifier: 0x0
19:42:37 201.279327 T:1957675568   DEBUG: OnKey: return (0xf00d) pressed, action is Select
19:42:37 201.281540 T:1957675568   DEBUG: Clearing cached fileitems [plugin://plugin.video.svtplay/?mode=pr&url=%2fvem-vet-mest]
19:42:37 201.282654 T:1957675568   DEBUG: CGUIMediaWindow::GetDirectory (plugin://plugin.video.svtplay/?mode=pr&url=%2fvem-vet-mest)
19:42:37 201.282852 T:1957675568   DEBUG:   ParentPath = [plugin://plugin.video.svtplay/?mode=a-o]
19:42:37 201.288177 T:1655698416   DEBUG: Thread JobWorker start, auto delete: true
19:42:37 201.289017 T:1655698416   DEBUG: StartScript - calling plugin SVT Play('plugin://plugin.video.svtplay/','2','?mode=pr&url=%2fvem-vet-mest')
19:42:37 201.289612 T:1777345520   DEBUG: Thread LanguageInvoker start, auto delete: false
19:42:37 201.289810 T:1655698416   DEBUG: WaitOnScriptResult - waiting on the SVT Play (id=5) plugin...
19:42:37 201.290176 T:1777345520    INFO: initializing python engine.
19:42:37 201.290344 T:1777345520   DEBUG: CPythonInvoker(5, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): start processing
19:42:37 201.693695 T:1777345520   DEBUG: -->Python Interpreter Initialized<--
19:42:37 201.694046 T:1777345520   DEBUG: CPythonInvoker(5, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): the source file to load is "/home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py"
19:42:37 201.694855 T:1777345520   DEBUG: CPythonInvoker(5, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): setting the Python path to /home/osmc/.kodi/addons/xbmc-svtplay-gotham:/home/osmc/.kodi/addons/script.module.parsedom/lib:/usr/lib/python2.7:/usr/lib/python2.7/plat-arm-linux-gnueabihf:/usr/lib/python2.7/lib-tk:/usr/lib/python2.7/lib-old:/usr/lib/python2.7/lib-dynload:/usr/local/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages/PILcompat:/usr/lib/python2.7/dist-packages/gtk-2.0
19:42:37 201.695023 T:1777345520   DEBUG: CPythonInvoker(5, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): entering source directory /home/osmc/.kodi/addons/xbmc-svtplay-gotham
19:42:37 201.695358 T:1777345520   DEBUG: CPythonInvoker(5, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): instantiating addon using automatically obtained id of "plugin.video.svtplay" dependent on version 2.14.0 of the xbmc.python api
19:42:37 201.788223 T:1957675568   DEBUG: ------ Window Init (DialogBusy.xml) ------
19:42:38 202.920242 T:1777345520   DEBUG: CommonFunctions-2.5.1
19:42:40 204.455597 T:1777345520   DEBUG: LocalizeStrings: no strings.po file exist at /home/osmc/.kodi/addons/xbmc-svtplay-gotham/resources/language/English, fallback to strings.xml
19:42:40 204.513336 T:1655698416   DEBUG: WaitOnScriptResult- plugin returned successfully
19:42:40 204.513962 T:1777345520    INFO: CPythonInvoker(5, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): script successfully run
19:42:40 204.553604 T:1957675568   DEBUG: Saving fileitems [plugin://plugin.video.svtplay/?mode=pr&url=%2fvem-vet-mest]
19:42:40 204.557327 T:1957675568   DEBUG:   -- items: 49, sort method: 0, ascending: false
19:42:40 204.569748 T:1957675568   DEBUG: RunQuery took 3 ms for 138 items query: SELECT  files.strFilename, files.playCount,  bookmark.timeInSeconds, bookmark.totalTimeInSeconds FROM files  LEFT JOIN bookmark ON    files.idFile = bookmark.idFile AND bookmark.type = 1  WHERE files.idPath=7
19:42:40 204.620911 T:1777345520    INFO: Python script stopped
19:42:40 204.621262 T:1777345520   DEBUG: Thread LanguageInvoker 1777345520 terminating
19:42:40 204.628311 T:1777345520   DEBUG: Thread BackgroundLoader start, auto delete: false
19:42:40 204.685730 T:1957675568   DEBUG: ------ Window Deinit (DialogBusy.xml) ------
19:42:40 204.870819 T:1777345520   DEBUG: Thread BackgroundLoader 1777345520 terminating
19:42:44 208.048416 T:1957675568   DEBUG: Keyboard: scancode: 0x1c, sym: 0x000d, unicode: 0x0000, modifier: 0x0
19:42:44 208.187698 T:1957675568   DEBUG: OnKey: return (0xf00d) pressed, action is Select
19:42:44 208.188690 T:1957675568   DEBUG: CGUIMediaWindow::GetDirectory (plugin://plugin.video.svtplay/?mode=a-o)
19:42:44 208.188965 T:1957675568   DEBUG:   ParentPath = [plugin://plugin.video.svtplay/]
19:42:44 208.220001 T:1957675568   DEBUG: Loading items: 445, directory: plugin://plugin.video.svtplay/?mode=a-o sort method: 0, ascending: false
19:42:44 208.407059 T:1777345520   DEBUG: Thread BackgroundLoader start, auto delete: false
19:42:45 209.254211 T:1777345520   DEBUG: Thread BackgroundLoader 1777345520 terminating
19:42:46 210.917847 T:1957675568   DEBUG: Keyboard: scancode: 0x1c, sym: 0x000d, unicode: 0x0000, modifier: 0x0
19:42:47 211.038513 T:1957675568   DEBUG: OnKey: return (0xf00d) pressed, action is Select
19:42:47 211.040695 T:1957675568   DEBUG: Clearing cached fileitems [plugin://plugin.video.svtplay/?mode=pr&url=%2fvem-vet-mest]
19:42:49 213.233566 T:1957675568   DEBUG: CGUIMediaWindow::GetDirectory (plugin://plugin.video.svtplay/?mode=pr&url=%2fvem-vet-mest)
19:42:49 213.233810 T:1957675568   DEBUG:   ParentPath = [plugin://plugin.video.svtplay/?mode=a-o]
19:42:49 213.237488 T:1655698416   DEBUG: StartScript - calling plugin SVT Play('plugin://plugin.video.svtplay/','3','?mode=pr&url=%2fvem-vet-mest')
19:42:49 213.238449 T:1777345520   DEBUG: Thread LanguageInvoker start, auto delete: false
19:42:49 213.238724 T:1777345520    INFO: initializing python engine.
19:42:49 213.238846 T:1777345520   DEBUG: CPythonInvoker(6, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): start processing
19:42:49 213.238998 T:1655698416   DEBUG: WaitOnScriptResult - waiting on the SVT Play (id=6) plugin...
19:42:49 213.616089 T:1777345520   DEBUG: -->Python Interpreter Initialized<--
19:42:49 213.616516 T:1777345520   DEBUG: CPythonInvoker(6, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): the source file to load is "/home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py"
19:42:49 213.617783 T:1777345520   DEBUG: CPythonInvoker(6, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): setting the Python path to /home/osmc/.kodi/addons/xbmc-svtplay-gotham:/home/osmc/.kodi/addons/script.module.parsedom/lib:/usr/lib/python2.7:/usr/lib/python2.7/plat-arm-linux-gnueabihf:/usr/lib/python2.7/lib-tk:/usr/lib/python2.7/lib-old:/usr/lib/python2.7/lib-dynload:/usr/local/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages/PILcompat:/usr/lib/python2.7/dist-packages/gtk-2.0
19:42:49 213.617950 T:1777345520   DEBUG: CPythonInvoker(6, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): entering source directory /home/osmc/.kodi/addons/xbmc-svtplay-gotham
19:42:49 213.618683 T:1777345520   DEBUG: CPythonInvoker(6, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): instantiating addon using automatically obtained id of "plugin.video.svtplay" dependent on version 2.14.0 of the xbmc.python api
19:42:49 213.737167 T:1957675568   DEBUG: ------ Window Init (DialogBusy.xml) ------
19:42:50 214.741623 T:1777345520   DEBUG: CommonFunctions-2.5.1
19:42:52 216.052383 T:1777345520   DEBUG: LocalizeStrings: no strings.po file exist at /home/osmc/.kodi/addons/xbmc-svtplay-gotham/resources/language/English, fallback to strings.xml
19:42:52 216.968079 T:1655698416   DEBUG: WaitOnScriptResult- plugin returned successfully
19:42:52 216.968597 T:1777345520    INFO: CPythonInvoker(6, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): script successfully run
19:42:52 217.008545 T:1957675568   DEBUG: Saving fileitems [plugin://plugin.video.svtplay/?mode=pr&url=%2fvem-vet-mest]
19:42:52 217.012833 T:1957675568   DEBUG:   -- items: 49, sort method: 0, ascending: false
19:42:53 217.026657 T:1957675568   DEBUG: RunQuery took 3 ms for 138 items query: SELECT  files.strFilename, files.playCount,  bookmark.timeInSeconds, bookmark.totalTimeInSeconds FROM files  LEFT JOIN bookmark ON    files.idFile = bookmark.idFile AND bookmark.type = 1  WHERE files.idPath=7
19:42:53 217.068680 T:1777345520    INFO: Python script stopped
19:42:53 217.068985 T:1777345520   DEBUG: Thread LanguageInvoker 1777345520 terminating
19:42:53 217.088699 T:1777345520   DEBUG: Thread BackgroundLoader start, auto delete: false
19:42:53 217.188110 T:1957675568   DEBUG: ------ Window Deinit (DialogBusy.xml) ------
19:42:53 217.298416 T:1777345520   DEBUG: Thread BackgroundLoader 1777345520 terminating
19:42:53 217.906616 T:1957675568   DEBUG: Keyboard: scancode: 0x6c, sym: 0x0112, unicode: 0x0000, modifier: 0x0
19:42:53 217.906967 T:1957675568   DEBUG: OnKey: down (0xf081) pressed, action is Down
19:42:53 217.916321 T:1655698416    INFO: easy_aquire - Created session to http://www.svt.se
19:42:54 218.092880 T:1655698416   DEBUG: CCurlFile::GetMimeType - http://www.svt.se/cachable_image/1458002341000/svts/article7155511.svt/ALTERNATES/extralarge/default_title -> image/jpeg
19:42:54 218.094254 T:1655698416   DEBUG: CFileCache::Open - opening <cachable_image/1458002341000/svts/article7155511.svt/ALTERNATES/extralarge/default_title> using cache
19:42:54 218.094971 T:1655698416   DEBUG: CurlFile::Open(0x4344670) http://www.svt.se/cachable_image/1458002341000/svts/article7155511.svt/ALTERNATES/extralarge/default_title
19:42:54 218.253983 T:1777345520   DEBUG: Thread FileCache start, auto delete: false
19:42:54 218.271881 T:1777345520    INFO: CFileCache::Process - Hit eof.
19:42:54 218.322067 T:1655698416    INFO: easy_aquire - Created session to http://www.svt.se
19:42:54 218.396591 T:1655698416   DEBUG: CCurlFile::GetMimeType - http://www.svt.se/cachable_image/1458002341000/svts/article7155511.svt/ALTERNATES/extralarge/default_title -> image/jpeg
19:42:54 218.437668 T:1777345520   DEBUG: Thread FileCache 1777345520 terminating
19:42:54 218.437973 T:1655698416   DEBUG: Caching image 'http://www.svt.se/cachable_image/1458002341000/svts/article7155511.svt/ALTERNATES/extralarge/default_title' to 'd/d6dcbe31.jpg':
19:42:54 218.554520 T:1957675568   DEBUG: Keyboard: scancode: 0x1c, sym: 0x000d, unicode: 0x0000, modifier: 0x0
19:42:54 218.688400 T:1957675568   DEBUG: OnKey: return (0xf00d) pressed, action is Select
19:42:54 218.708710 T:1957675568   DEBUG: OnPlayMedia plugin://plugin.video.svtplay/?url=%2Fvideo%2F7155505%2Fvem-vet-mest%2Fvem-vet-mest-sasong-16-avsnitt-48&mode=video
19:42:54 218.729233 T:1957675568   DEBUG: CAnnouncementManager - Announcement: OnAdd from xbmc
19:42:54 218.729584 T:1957675568   DEBUG: GOT ANNOUNCEMENT, type: 2, from xbmc, message OnAdd
19:42:54 218.739288 T:1957675568   DEBUG: StartScript - calling plugin SVT Play('plugin://plugin.video.svtplay/','4','?url=%2Fvideo%2F7155505%2Fvem-vet-mest%2Fvem-vet-mest-sasong-16-avsnitt-48&mode=video')
19:42:54 218.740280 T:1777345520   DEBUG: Thread LanguageInvoker start, auto delete: false
19:42:54 218.740540 T:1957675568   DEBUG: WaitOnScriptResult - waiting on the SVT Play (id=7) plugin...
19:42:54 218.741058 T:1777345520    INFO: initializing python engine.
19:42:54 218.741196 T:1777345520   DEBUG: CPythonInvoker(7, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): start processing
19:42:54 218.840759 T:1655698416   DEBUG: cached image 'special://masterprofile/Thumbnails/d/d6dcbe31.jpg' size 960x540
19:42:54 218.844635 T:1655698416   DEBUG: COMXCoreComponent::Initialize OMX.broadcom.image_encode input port 340 output port 341 m_handle 0x445b6f8
19:42:54 218.847656 T:1655698416   DEBUG: COMXCoreComponent::AllocInputBuffers component(OMX.broadcom.image_encode) - port(340), nBufferCountMin(1), nBufferCountActual(1), nBufferSize(2088960), nBufferAlignmen(16)
19:42:54 218.851089 T:1655698416   DEBUG: COMXCoreComponent::AllocOutputBuffers component(OMX.broadcom.image_encode) - port(341), nBufferCountMin(1), nBufferCountActual(1), nBufferSize(522240) nBufferAlignmen(16)
19:42:54 218.886887 T:1655698416   DEBUG: COMXImageEnc::CreateThumbnailFromSurface : special://masterprofile/Thumbnails/d/d6dcbe31.jpg width 960 height 540
19:42:54 218.892929 T:1655698416   DEBUG: COMXCoreComponent::Deinitialize : OMX.broadcom.image_encode handle 0x445b6f8
19:42:55 219.160294 T:1777345520   DEBUG: -->Python Interpreter Initialized<--
19:42:55 219.160660 T:1777345520   DEBUG: CPythonInvoker(7, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): the source file to load is "/home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py"
19:42:55 219.161270 T:1777345520   DEBUG: CPythonInvoker(7, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): setting the Python path to /home/osmc/.kodi/addons/xbmc-svtplay-gotham:/home/osmc/.kodi/addons/script.module.parsedom/lib:/usr/lib/python2.7:/usr/lib/python2.7/plat-arm-linux-gnueabihf:/usr/lib/python2.7/lib-tk:/usr/lib/python2.7/lib-old:/usr/lib/python2.7/lib-dynload:/usr/local/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages:/usr/lib/python2.7/dist-packages/PILcompat:/usr/lib/python2.7/dist-packages/gtk-2.0
19:42:55 219.161453 T:1777345520   DEBUG: CPythonInvoker(7, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): entering source directory /home/osmc/.kodi/addons/xbmc-svtplay-gotham
19:42:55 219.161667 T:1777345520   DEBUG: CPythonInvoker(7, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): instantiating addon using automatically obtained id of "plugin.video.svtplay" dependent on version 2.14.0 of the xbmc.python api
19:42:56 220.261078 T:1957675568   DEBUG: DialogProgress::Open called
19:42:56 220.261353 T:1957675568   DEBUG: ------ Window Init (DialogProgress.xml) ------
19:42:56 220.261505 T:1957675568    INFO: Loading skin file: DialogProgress.xml, load type: KEEP_IN_MEMORY
19:42:56 220.291122 T:1777345520   DEBUG: CommonFunctions-2.5.1
19:42:57 221.130127 T:1957675568   DEBUG: WaitOnScriptResult- plugin returned successfully
19:42:57 221.153366 T:1957675568   DEBUG: Loading settings for index_6_av.m3u8?&e=82d461566017da20&alt=http%3A%2F%2Fswitcher.cdn.svt.se%2F1369718-048A.m3u8
19:42:57 221.169724 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers(index_6_av.m3u8?&e=82d461566017da20&alt=http%3A%2F%2Fswitcher.cdn.svt.se%2F1369718-048A.m3u8)
19:42:57 221.169922 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: system rules
19:42:57 221.170212 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: matches rule: system rules
19:42:57 221.170364 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: mms/udp
19:42:57 221.170532 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: lastfm/shout
19:42:57 221.170685 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: rtmp
19:42:57 221.170914 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: rtsp
19:42:57 221.171051 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: streams
19:42:57 221.171143 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: dvd
19:42:57 221.171310 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: dvdimage
19:42:57 221.171417 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: sdp/asf
19:42:57 221.171555 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: nsv
19:42:57 221.171753 T:1957675568   DEBUG: CPlayerSelectionRule::GetPlayers: considering rule: radio
19:42:57 221.171890 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers: matched 0 rules with players
19:42:57 221.172073 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers: adding videodefaultplayer (1)
19:42:57 221.172195 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers: for video=1, audio=0
19:42:57 221.172256 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers: for video=1, audio=1
19:42:57 221.172318 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers: adding player: DVDPlayer (1)
19:42:57 221.172607 T:1957675568   DEBUG: CPlayerCoreFactory::GetPlayers: added 1 players
19:42:57 221.186600 T:1957675568   DEBUG: Radio UECP (RDS) Processor - new CDVDRadioRDSData
19:42:57 221.186951 T:1957675568  NOTICE: DVDPlayer: Opening: index_6_av.m3u8?&e=82d461566017da20&alt=http%3A%2F%2Fswitcher.cdn.svt.se%2F1369718-048A.m3u8
19:42:57 221.187164 T:1957675568 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
19:42:57 221.187424 T:1957675568   DEBUG: CMMALRenderer::CMMALRenderer
19:42:57 221.187546 T:1957675568   DEBUG: CRenderManager::UpdateDisplayLatency - Latency set to 0 msec
19:42:57 221.187622 T:1957675568   DEBUG: CMMALRenderer::UnInitMMAL pool((nil))
19:42:57 221.187698 T:1957675568   DEBUG: CMMALRenderer::PreInit
19:42:57 221.188019 T:1749545968   DEBUG: Thread DVDPlayer start, auto delete: false
19:42:57 221.189178 T:1749545968  NOTICE: Creating InputStream
19:42:57 221.189972 T:1749545968   ERROR: CDVDPlayer::OpenInputStream - error opening [index_6_av.m3u8?&e=82d461566017da20&alt=http%3A%2F%2Fswitcher.cdn.svt.se%2F1369718-048A.m3u8]
19:42:57 221.190186 T:1749545968  NOTICE: CDVDPlayer::OnExit()
19:42:57 221.190338 T:1749545968   DEBUG: OnPlayBackStopped: play state was 1, starting 1
19:42:57 221.190460 T:1749545968   DEBUG: Thread DVDPlayer 1749545968 terminating
19:42:57 221.190567 T:1957675568   DEBUG: OnPlayBackStopped: play state was 3, starting 0
19:42:57 221.191055 T:1957675568   DEBUG: CAnnouncementManager - Announcement: OnStop from xbmc
19:42:57 221.191391 T:1957675568   DEBUG: GOT ANNOUNCEMENT, type: 1, from xbmc, message OnStop
19:42:57 221.191895 T:1957675568   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.svtplay/?url=%2Fvideo%2F7155505%2Fvem-vet-mest%2Fvem-vet-mest-sasong-16-avsnitt-48&mode=video]
19:42:57 221.192032 T:1957675568   DEBUG: Playlist Player: no more playable items... aborting playback
19:42:57 221.193008 T:1749545968   DEBUG: Thread BackgroundLoader start, auto delete: false
19:42:57 221.208221 T:1957675568  NOTICE: CDVDPlayer::CloseFile()
19:42:57 221.208435 T:1957675568  NOTICE: DVDPlayer: waiting for threads to exit
19:42:57 221.208817 T:1957675568  NOTICE: DVDPlayer: finished waiting
19:42:57 221.209030 T:1957675568   DEBUG: CMMALRenderer::UnInitMMAL pool((nil))
19:42:57 221.209335 T:1957675568  NOTICE: CDVDPlayer::CloseFile()
19:42:57 221.209579 T:1957675568  NOTICE: DVDPlayer: waiting for threads to exit
19:42:57 221.209793 T:1957675568  NOTICE: DVDPlayer: finished waiting
19:42:57 221.210297 T:1957675568   DEBUG: CMMALRenderer::UnInitMMAL pool((nil))
19:42:57 221.214432 T:1957675568   DEBUG: Radio UECP (RDS) Processor - delete ~CDVDRadioRDSData
19:42:57 221.216461 T:1957675568   DEBUG: ------ Window Deinit (DialogProgress.xml) ------
19:42:57 221.368652 T:1749545968   DEBUG: Thread BackgroundLoader 1749545968 terminating
19:43:06 230.669754 T:1777345520 WARNING: Attempt to use invalid handle 4
19:43:06 230.670639 T:1777345520    INFO: CPythonInvoker(7, /home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py): script successfully run
19:43:06 230.789398 T:1777345520    INFO: Python script stopped
19:43:06 230.789948 T:1777345520   DEBUG: Thread LanguageInvoker 1777345520 terminating
^C
]0;osmc@osmc: ~osmc@osmc:~$ exit
exit

Script done on Wed Mar 16 19:43:11 2016`

**Comment by linqcan** [comment]
Thanks for the feedback!
I tried all programs mentioned here before pushing 4.0.22 and had no
issues. Will hopefully be able to look at this today.

**Comment by pacmyc** [comment]
No problem, and thank you for developing this beautiful plugin.. I use it everyday so I'll assist you any way I can. It would be interesting to know how many people are using this plugin.

**Comment by ToSeX** [comment]
A question...  Witch version of kodi did you test this app in? 
:-)
Br
Tobbe

**Comment by ToSeX** [comment]
A question...  Witch version of kodi did you test this app in? 
What I mean is that it worked fine before we had the last Kodi patch... 
:-)
Br
Tobbe

**Comment by pacmyc** [comment]
February 2016.02-3, Kodi 16.0 (compiled feb 21 2016)

**Comment by linqcan** [comment]
Do you, who have issues, have the "Set bandwidth manually" setting enabled?
If so, please turn it off and try again.

**Comment by pacmyc** [comment]
Nope, only "Don't use avc 1.77.30 streams" enabled..

**Comment by linqcan** [comment]
That setting gives the same issue. Please disable both and try again.

The issue is that most of the URL for the stream is missing. See the snippet here:

```
ERROR: CDVDPlayer::OpenInputStream - error opening [index_6_av.m3u8?&e=82d461566017da20&alt=http%3A%2F%2Fswitcher.cdn.svt.se%2F1369718-048A.m3u8]
```

There should be a `http://svtplay1n-f.akamaihd.net/i/world/open...` part before the m3u8 playlist item.

I am looking into why they (the settings) are causing this issue now, suspecting that the m3u8s has changed.

**Comment by pacmyc** [comment]
Now it works after unchecking "Don't use avc 1.77.30 streams".. Thanks!

**Comment by linqcan** [comment]
Good to hear!

Still looking into the root cause. But please disable both "Don't use avc 1.77.30 streams" and "Set bandwidth manually" as a workaround until a proper fix exists.

**Comment by pacmyc** [comment]
Allright,, let me know if you'd like me to do any testing.

**Comment by linqcan** [comment]
If possible, please try the latest version 80b6d241adb046c105ceb63d637da3f7f3684f1a

It should now work to use the settings again.

**Comment by linqcan** [comment]
Has anyone managed to test v4.0.23 yet?
My tests looks OK so I am planning to submit it to Kodi if there are no objections.

**Comment by rikardfalkeborn** [comment]
I've only tested briefly, but so far I've seen no issues opening streams so far with v4.0.23 (Osmc on Raspberry Pi), including streams that failed in v4.0.21. Good job!

**Comment by ToSeX** [comment]
Can you test the "Kategori" shortcut so it workes there?

**Comment by rikardfalkeborn** [comment]
Sorry for the late response. "Kategorier" does not work with 4.0.21 or 4.0.23. But that's probably a separate issue?
The log shows this:

```
20:08:24 875.497681 T:2814899184   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.TypeError'>
                                            Error Contents: 'NoneType' object is not iterable
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py", line 354, in <module>
                                                viewCategories()
                                              File "/home/osmc/.kodi/addons/xbmc-svtplay-gotham/default.py", line 105, in viewCategories
                                                for category in categories:
                                            TypeError: 'NoneType' object is not iterable
                                            -->End of Python script error report<--
20:08:24 875.868713 T:3025093168   ERROR: GetDirectory - Error getting plugin://plugin.video.svtplay/?mode=categories
20:08:24 875.875427 T:3025093168   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.svtplay/?mode=categories) failed
```

**Comment by linqcan** [comment]
Not related to this issue, but related to the current open issue. You can
comment in that one for traceability. All these new failures (again, not
the issue this thread was originally about) are related to SVT moving to
use React for generating the HTML. Can possibly make things really messy in
the future for scrapers like this plug in.
